src/attributes.jl

```diff
@@ -1138,6 +1138,27 @@ indices of type `F`-in-`S` in the model (i.e., of length equal to the value of
 """
 struct ListOfConstraintIndices{F,S} <: AbstractModelAttribute end
 
+function _may_have_been_added(
+    model::ModelLike,
+    ::Type{F},
+    ::Type{S}
+) where {F, S}
+    return MOI.supports_constraint(model, F, S) ||
+        (S === MOI.VariableIndex && MOI.supports_add_constrained_variable(model, S)) ||
+        (S === MOI.VectorOfVariables && MOI.supports_add_constrained_variables(model, S))
+end
+
+function get(
+    model::ModelLike,
+    attr::ListOfConstraintIndices{F,S},
+) where {F,S}
+    if _may_have_been_added(model, F, S)
+        error("Not implemented")
```

Comment:
Why have this error instead of just returning the empty vector? Same with erroring instead of returning `0` below.

Reply:
The error should be improved but basically, the MOI implementation is responsible for implementing this in case the constraint type is supported. Returning an empty vector would be incorrect in case the MOI implementation does not implement it. I don't see situation where this fallback would be correct. It would mean that an MOI implementation supports a constraint but knows that there is always going to be zero constraints of that type